x-pack/plugins/apm/server/routes/alerts/register_transaction_duration_alert_type.ts

```diff
@@ -171,15 +180,28 @@ export function registerTransactionDurationAlertType({
         return {};
       }
 
-      const { latency } = response.aggregations;
-
-      const transactionDuration =
-        'values' in latency ? Object.values(latency.values)[0] : latency?.value;
-
       // Converts threshold to microseconds because this is the unit used on transactionDuration
       const thresholdMicroseconds = ruleParams.threshold * 1000;
 
-      if (transactionDuration && transactionDuration > thresholdMicroseconds) {
+      const triggeredEnvironmentDurations =
+        response.aggregations.environments.buckets
+          .map(({ key: environment, latency }) => {
+            const transactionDuration =
+              'values' in latency
+                ? Object.values(latency.values)[0]
+                : latency?.value;
```

Comment:
If you use `"keyed": false` in the [percentile agg](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-percentile-aggregation.html#_keyed_response_6), you'll get an array back instead of an object. 

Can you try this (TS might complain over it but worth a shot):

```suggestion
            const transactionDuration = ruleParams.aggregationType === 'avg' ? latency?.value : latency.values[0].value;
```

Reply:
Good call on using `"keyed": false`.

Unfortunately the condition `ruleParams.aggregationType === 'avg'` does not narrow the latency type in TS. However I think achieved a more readable solution which still uses the `in` operator, by naming the aggs differenty:
https://github.com/elastic/kibana/pull/143238/files#diff-0d9fd3262d21e6bbd4fa2fb0fa06a0e54bdb75a8e73fdccf41190aace454a128R193